### PR TITLE
Fix json schema for Qasm qobj snapshot

### PR DESCRIPTION
**Pull request by chriseclectic**
### Summary

PR #2164 changed the qasm qobj field for snapshots from `"type"` to `"snapshot_type"`, but didn't update the JSON schema so circuits with snapshots fail validation. This PR updates the schema to correct format.

### CODE REVIEW

File: qiskit/schemas/qobj_schema.json [hunk 1]
```diff
@@ -639,7 +639,7 @@
                     },
                     "required": [
                         "label",
-                        "type"
+                        "snapshot_type"
```
Comment (maddy-tod) [inline qiskit/schemas/qobj_schema.json]: Should the `type` above this (line 636) also be updated to `snapshot_type`?

**Comment by maddy-tod** [comment]
Looks good - this will fix #2193

File: qiskit/schemas/qobj_schema.json (hunk 1 above)
Comment (chriseclectic) [reply qiskit/schemas/qobj_schema.json]: Yes I think it should!

**Comment by chriseclectic** [comment]
After looking at the snapshot fields more closely I changed them too. It now checks that if a snapshot has params, qubits, memory, or registers that they are an array. These fields may be used for certain types of advanced snapshot functions.

**Pull request merged**